pkg/internal/limitrange/transformer.go

```diff
@@ -45,33 +45,45 @@ func NewTransformer(ctx context.Context, namespace string, lister corev1listers.
 
 		// The assumption here is that the min, max, default, ratio have already been
 		// computed if there is multiple LimitRange to satisfy the most (if we can).
-		// Count the number of containers (that we know) in the Pod.
+		// Count the number of step containers in the Pod.
 		// This should help us find the smallest request to apply to containers
-		nbContainers := len(p.Spec.Containers)
+		nbStepContainers := 0
+		for _, c := range p.Spec.Containers {
+			if pod.IsContainerStep(c.Name) {
```

Comment:
do we need to check for the running containers or any other specific state (not-running) of the container here? or the state of container does not matter at this point since they will be spawned with the configuration being determined here?

Reply:
alright, so this is only used in pod builder while creating a pod so the state is not relevant here ...

Reply:
The transform happens before the pod spec is applied so the state of the container does not matter at this point. We use the number of step containers to help choose the optimal request size at configuration generation time.